Update Amoebeye.cs

Mellozx opened the pull request:
Fixed amoebeye only facing left

Mellozx committed:
Update Amoebeye.cs

Fixed amoebeye only facing left

Tyfyter commented:
This appears to be a substantial change to behavior, will have to review more thoroughly than just viewing the core allows for.

Mellozx committed:
Reworked Amoebeye AI completely

-- Reworks:
Amoebeye will float towards the player in a hovering state like before, however
it no longer gets enraged instead it will have a Dive Attack every couple of seconds, floating upwards then rushing towards the player for a strike
---
It can only have one Amoebeye projectile alive at a time unless 25 seconds pass without the projectile dying, it will spawn another, this is to balance its ability against the player. If the player kills a projectile it will spawn another in 2-3 seconds
---
Projectile AI and behavior is unchanged
---
Added comments to explain what my code does

Mellozx commented:
Rewrote the AI to make it fit better, the actual fix is to only switch aistyle to none but I decided to rework the NPC as a whole to give it a more unique style rather than just enraging at half health

Tyfyter commented:
The health has never been a factor in whether or not it's enraged, and upon further inspection it looks like you initially made it perpetually enraged.
